Lib/test/test_asyncio/utils.py

```diff
@@ -557,9 +558,9 @@ def close_loop(loop):
                 pass
             else:
                 if isinstance(watcher, asyncio.ThreadedChildWatcher):
-                    threads = list(watcher._threads.values())
-                    for thread in threads:
-                        thread.join()
+                    # Wait for subprocess to finish, but not forever
+                    for thread in list(watcher._threads.values()):
+                        thread.join(timeout=support.SHORT_TIMEOUT)
```

Comment:
You could terminate the child processes from `watcher._threads.key()`?

Reply:
Oh, that would be a useful thing for `_join_threads()` to do -- just `os.kill(pid, 9)`. (Or killpg?)

Reply:
I think it would be useful in the cpython tests but would be an unexpected new behaviour of this were done in ThreadedChildWatcher

Reply:
That's fair. So let's do it here. Do you feel like adding a proposed diff to this PR?

Reply:
I've been thinking about this and I think it's better to get a TimeoutError in the test and let the subprocess leak so it can be fixed in subsequent test runs rather than papering over it here by killing them

Reply:
```suggestion
                        thread.join(timeout=support.SHORT_TIMEOUT)
                        if thread.is_alive():
                            raise RuntimeError(f"thread {thread} still alive due to subprocess still running")
```